hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -1826,6 +1828,19 @@ private Map<byte[], List<HStoreFile>> doClose(boolean abort, MonitoredTask statu
         LOG.error(msg);
         if (canAbort) {
           // If we failed to acquire the write lock, abort the server
+          // print all thread stacks which are still holding locks and are the cause of RS abort
+          for (Map.Entry<Thread, Boolean> rslocks : intrRegionLockHolders.entrySet()) {
+            Thread thread = rslocks.getKey();
+            LOG.warn(
+              "Aborting RS.Thread still holding lock: {} , interruptible: {} , Stack trace: {}",
+              thread, rslocks.getValue(), Threads.printStackTrace(thread));
+          }
+          for (Map.Entry<Thread, Boolean> rslocks : nonintrRegionLockHolders.entrySet()) {
```

Comment:
I am thinking of how we can improve this piece of code. This seems very redundant.